docs/src/modules/java/pages/dev-best-practices.adoc

```diff
@@ -47,3 +47,186 @@ Deciding how many components and concepts to fit into a single service can be co
+
+Events modelling is a key part of the system design. A consumer that have all the necessary information in the event can be more independent and less error-prone. Of course a balance must be found between the size of the event and simplicity of its processing.
+
+
+[source,java,indent=0]
+.{sample-base-url}/event-sourced-counter-brokers/src/main/java/counter/domain/CounterEvent.java[CounterEvent.java]
+----
+include::example$event-sourced-counter-brokers/src/main/java/counter/domain/CounterEvent.java[tag=events-enrichment]
+----
+<1> `ValueMultiplied` event contains not only delta information under `multiplier` field but also pre-calculated `currentValue` of the counter.
+
+The updated version of the `CounterByValueUpdater` can be again idempotent.
+
+[source,java,indent=0]
+.{sample-base-url}/event-sourced-counter-brokers/src/main/java/counter/application/CounterByValueViewEnrichment.java[CounterByValueViewEnrichment.java]
+----
+include::example$event-sourced-counter-brokers/src/main/java/counter/application/CounterByValueViewEnrichment.java[tag=events-enrichment]
+----
+<1> Using pre-calculated `currentValue` from the event.
+
+==== Benefits
+ - Enables idempotent processing of enriched events.
+ - Reduces coupling between producers and consumers.
+
+==== Challenges
+ - Increases event size, requiring a balance between richness and efficiency.
+ - Requires careful schema design and potential for schema evolution challenges
+
+=== Sequence number tracking
+
+For cases when events enrichment is not possible or not desired, a sequence number tracking can be used. The idea is to keep track of the sequence number of the last processed event and ignore any events with a sequence number lower or equal than the last processed one.
+
+NOTE: A monotonically increased sequence number is available only when consuming updates from Akka components like Event Sourced Entity, Key Value Entity, or another Akka service. The sequence number is *not globally unique*, but unique per entity instance.
+
+NOTE: Akka View component has a built-in support for sequence number tracking. {sample-base-url}/event-sourced-counter-brokers/src/test/java/counter/application/CounterByValueViewTest.java[CounterByValueViewTest] demonstrates how it can be verified.
+
+Let's assume that we want to populate a view storage outside Akka ecosystem. To focus on the deduplication aspect, the following snippet shows the in-memory implementation of the `CounterStore`.
+
+[source,java,indent=0]
+.{sample-base-url}/event-sourced-counter-brokers/src/main/java/counter/application/CounterStore.java[CounterStore.java]
+----
+include::example$event-sourced-counter-brokers/src/main/java/counter/application/CounterStore.java[tag=seq-tracking]
+----
+<1> A view model keeps track of the last processed sequence number.
```

Comment:
"view model"? Is that from previous example?

Reply:
changed to read model